integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/securevault/test/JMSTransportSecureVaultTest.java

```diff
@@ -0,0 +1,185 @@
+        targetFile = new File(targetFileLocation);
+        serverConfigurationManager.applyConfigurationWithoutRestart(srcFile, targetFile, false);
+
+        serverConfigurationManager.copyToComponentLib(new File(getClass().getResource(JAR_DIR + ACTIVEMQ_BROKER)
+                .getPath()));
+        serverConfigurationManager.copyToComponentLib(new File(getClass().getResource(JAR_DIR + ACTIVEMQ_CLIENT)
+                .getPath()));
+        serverConfigurationManager.copyToComponentLib(new File(getClass().getResource(JAR_DIR
+                + GERONIMO_J2EE_MANAGEMENT).getPath()));
+        serverConfigurationManager.copyToComponentLib(new File(getClass().getResource(JAR_DIR + GERONIMO_JMS)
+                .getPath()));
+        serverConfigurationManager.copyToComponentLib(new File(getClass().getResource(JAR_DIR + HAWTBUF).getPath()));
+        serverConfigurationManager.copyToComponentLib(new File(getClass().getResource(JAR_DIR + ACTIVEMQ_KAHADB_STORE)
+                .getPath()));
+        serverConfigurationManager.copyToComponentLib(new File(getClass().getResource(JAR_DIR + SLF4J_API).getPath()));
+        serverConfigurationManager.copyToComponentLib(new File(getClass().getResource(JAR_DIR + ACTIVEIO_CORE)
+                .getPath()));
+
+        serverConfigurationManager.restartGracefully();
+        super.init();
+        loadESBConfigurationFromClasspath("artifacts" + File.separator + "ESB" + File.separator +
+                "jms" + File.separator + "securevault" + File.separator + SYNAPSE_CONFIG_FILE);
+        isProxyDeployed(PROXY_NAME);
+    }
+
+    private JMSBrokerConfiguration getJMSBrokerConfiguration() {
+        return JMSBrokerConfigurationProvider.getInstance().getBrokerConfiguration();
+    }
+
+    /**
+     * This test will test the JMS transport functionality (with password encrypted with SecureVault):
+     * A user with username "system" and password "manager" is introduced to the broker here, and the axis2.xml
+     * also includes this configuration, where the password ("manager") has been encrypted with SecureVault.
+     * Thus an exception (SecurityException) thrown while connecting to send would indicate a decryption failure.
+     */
+    @Test(groups = "wso2.esb", description = "Test JMS transport using parameters secured using SecureVault")
+    public void testJMSTransportWithSecureVault() throws InterruptedException {
+        try {
+            activeMQBroker = new JMSBroker("localhost", getJMSBrokerConfiguration());
+            simpleAuthenticationPlugin = new SimpleAuthenticationPlugin();
+            simpleAuthenticationPlugin.setAnonymousAccessAllowed(false);
+            List<AuthenticationUser> users = new ArrayList<AuthenticationUser>();
+            users.add(new AuthenticationUser("system", "manager", "users,admins"));
+            simpleAuthenticationPlugin.setUsers(users);
+            activeMQBroker.setPluginsAndStart(new BrokerPlugin[]{simpleAuthenticationPlugin});
```

Comment:
We continue even when the broker fails to start. Adding an assert might help to debug the test failures.